streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -172,7 +172,7 @@
      * </pre>
      */
     public enum State {
-        CREATED(1, 2, 3), RUNNING(2, 3), REBALANCING(1, 2, 3), PENDING_SHUTDOWN(4), NOT_RUNNING;
+        CREATED(1, 2, 3), REBALANCING(1, 2, 3), RUNNING(1, 2, 3), PENDING_SHUTDOWN(4), NOT_RUNNING;
```

Comment:
It used to be that we immediately went to RUNNING. Then when all threads were up, we set the state to RUNNING again. I've now changed the logic to start at CREATED instead. That should be easier to understand and we don't go to RUNNING twice.